probe/endpoint/procspy/proc.go

```diff
@@ -34,28 +34,25 @@ func walkProcPid(buf *bytes.Buffer, walker process.Walker) (map[uint64]Proc, err
 	walker.Walk(func(p process.Process) {
 		dirName := strconv.Itoa(p.PID)
 		fdBase := filepath.Join(procRoot, dirName, "fd")
-		fds, err := fs.ReadDir(fdBase)
-		if err != nil {
-			// Process is be gone by now, or we don't have access.
-			return
-		}
 
 		// Read network namespace, and if we haven't seen it before,
 		// read /proc/<pid>/net/tcp
-		err = fs.Lstat(filepath.Join(procRoot, dirName, "/ns/net"), &statT)
-		if err != nil {
+		if err := fs.Lstat(filepath.Join(procRoot, dirName, "/ns/net"), &statT); err != nil {
 			return
 		}
-
 		if _, ok := namespaces[statT.Ino]; !ok {
 			namespaces[statT.Ino] = struct{}{}
 			readFile(filepath.Join(procRoot, dirName, "/net/tcp"), buf)
-			readFile(filepath.Join(procRoot, dirName, "/net/tcp6"), buf)
 		}
 
+		fds, err := fs.ReadDirNames(fdBase)
+		if err != nil {
+			// Process is be gone by now, or we don't have access.
+			return
+		}
```

Comment:
The commit says this reduces the number of `stat`s we're doing, but here it looks like we've added one?